File: example/c/dense/generate_matrix/matrix012.c
/*****************************************************************************
*
* Rokko: Integrated Interface for libraries of eigenvalue decomposition
*
* Copyright (C) 2012-2019 by Rokko Developers https://github.com/t-sakashita/rokko
*
* Distributed under the Boost Software License, Version 1.0. (See accompanying
* file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
*
*****************************************************************************/

#include <rokko/rokko.h>
#include <rokko/utility/matrix012.h>
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[]) {
  struct rokko_eigen_matrix mat;
  int dim;

  dim = 10;
  rokko_eigen_matrix_construct(&mat, dim, dim, rokko_matrix_col_major);
  rokko_matrix012_generate_eigen_matrix(mat);
  rokko_eigen_matrix_print(mat);

  rokko_eigen_matrix_destruct(&mat);

  return 0;
}

File: example/fortran/dense/generate_matrix/matrix012.f90
!*****************************************************************************
!
! Rokko: Integrated Interface for libraries of eigenvalue decomposition
!
! Copyright (C) 2012-2019 by Rokko Developers https://github.com/t-sakashita/rokko
!
! Distributed under the Boost Software License, Version 1.0. (See accompanying
! file LICENSE_1_0.txt or copy at http://www.boost.org/license_1_0.txt)
!
!*****************************************************************************

program matrix012
  use rokko
  implicit none
  integer :: dim
  type(rokko_eigen_matrix) :: mat

  dim = 10
  print *,"dimension = ", dim

  call rokko_construct(mat, dim, dim, rokko_matrix_col_major)

  call rokko_matrix012_generate(mat)
  call rokko_print(mat)

  call rokko_destruct(mat)

end program matrix012
